examples/nlu-assistant/README.md

```diff
@@ -0,0 +1,515 @@
+
+##### 2.2 Creating your own custom model
+
+In this example, we are going to create a common model's configuration in terms of natural language processing. This is a stack of `sequential` layers consisting of an `embedding` layer, followed by a `LSTM` layer and finally a `dense` layer.
+
+Let's start by creating a new file called `lstm-model.ts` under **src/nlu/**.
+
+Using the power of [tfjs](https://www.tensorflow.org/js) you can define a new model as follows:
+
+**src/nlu/train-model.ts**
+
+```ts
+import { WordEmbeddingsConfig } from '@botonic/nlu/dist/types'
+import { WordEmbeddingsManager } from '@botonic/nlu/dist/embeddings/word-embeddings-manager'
+
+import { layers, sequential, Sequential } from '@tensorflow/tfjs-node'
+import { DenseLayerArgs } from '@tensorflow/tfjs-layers/dist/layers/core'
+import { LSTMLayerArgs, ModelCompileArgs } from '@tensorflow/tfjs-layers/dist'
+
+interface LayersConfig {
+  LSTM: LSTMLayerArgs
+  DENSE: DenseLayerArgs
+}
+
+export const LSTMModel = async (
+  maxSeqLen: number,
+  wordEmbeddingsConfig: WordEmbeddingsConfig,
+  layersConfig: LayersConfig,
+  modelCompileArgs: ModelCompileArgs
+): Promise<Sequential> => {
+  const wordEmbeddingsManager = await WordEmbeddingsManager.withConfig(
+    wordEmbeddingsConfig
+  )
+  const wordEmbeddingsMatrix = wordEmbeddingsManager.matrix
+  const model = sequential()
+  model.add(
+    layers.embedding({
+      inputDim: wordEmbeddingsMatrix.shape[0],
+      outputDim: wordEmbeddingsMatrix.shape[1],
+      inputLength: maxSeqLen,
+      trainable: false,
+      weights: [wordEmbeddingsMatrix],
+    })
+  )
+  model.add(layers.lstm(layersConfig.LSTM))
+  model.add(layers.dense(layersConfig.DENSE))
+  model.compile(modelCompileArgs)
+  model.summary()
+  return model
+}
+```
+
+We have defined here the interface `LayersConfig` to be able to tune the model easily from the outside. In this piece of we are:
+
+1.  Loading the matrix of word embeddings needed to initialize the embedding layer.
+    ```ts
+    const wordEmbeddingsManager = await WordEmbeddingsManager.withConfig(
+      wordEmbeddingsConfig
+    )
+    const wordEmbeddingsMatrix = wordEmbeddingsManager.matrix
+    ```
+2.  Place the embedding layer as the first layer in the stack:
+    ```ts
+    const model = sequential() // initializing the stack of layers
+    model.add(
+      layers.embedding({
+        inputDim: wordEmbeddingsMatrix.shape[0],
+        outputDim: wordEmbeddingsMatrix.shape[1],
+        inputLength: maxSeqLen,
+        trainable: false,
+        weights: [wordEmbeddingsMatrix],
+      })
+    )
+    ```
+    > **Note:** It's very important you define correctly these parameters for a correct functioning of the model.
```

Comment:
In the README view, all of these parameter explanations are not in a list format.